shell/platform/darwin/macos/framework/Source/FlutterOpenGLRenderer.mm

```diff
@@ -0,0 +1,193 @@
+}
+
+- (bool)makeCurrent {
+  if (!_openGLContext) {
+    return false;
+  }
+  [_openGLContext makeCurrentContext];
+  return true;
+}
+
+- (bool)clearCurrent {
+  [NSOpenGLContext clearCurrentContext];
+  return true;
+}
+
+- (bool)present {
+  if (!_openGLContext) {
+    return false;
+  }
+  [_flutterView present];
+  return true;
+}
+
+- (uint32_t)fboForFrameInfo:(const FlutterFrameInfo*)info {
+  CGSize size = CGSizeMake(info->size.width, info->size.height);
+  return [_flutterView frameBufferIDForSize:size];
+}
+
+- (NSOpenGLContext*)resourceContext {
+  if (!_resourceContext) {
+    NSOpenGLPixelFormatAttribute attributes[] = {
+        NSOpenGLPFAColorSize, 24, NSOpenGLPFAAlphaSize, 8, 0,
+    };
+    NSOpenGLPixelFormat* pixelFormat = [[NSOpenGLPixelFormat alloc] initWithAttributes:attributes];
+    _resourceContext = [[NSOpenGLContext alloc] initWithFormat:pixelFormat shareContext:nil];
+  }
+  return _resourceContext;
+}
+
+- (NSOpenGLContext*)openGLContext {
+  if (!_openGLContext) {
+    NSOpenGLContext* shareContext = [self resourceContext];
+    _openGLContext = [[NSOpenGLContext alloc] initWithFormat:shareContext.pixelFormat
+                                                shareContext:shareContext];
+  }
+  return _openGLContext;
+}
+
+- (bool)makeResourceCurrent {
+  [self.resourceContext makeCurrentContext];
+  return true;
+}
+
+- (void)clearResourceContext {
+  _resourceContext = nil;
+}
+
+#pragma mark - FlutterTextureRegistrar
+
+- (BOOL)populateTextureWithIdentifier:(int64_t)textureID
+                        openGLTexture:(FlutterOpenGLTexture*)openGLTexture {
+  return [_textures[@(textureID)] populateTexture:openGLTexture];
+}
+
+- (int64_t)registerTexture:(id<FlutterTexture>)texture {
+  FlutterExternalTextureGL* FlutterTexture =
+      [[FlutterExternalTextureGL alloc] initWithFlutterTexture:texture];
+  int64_t textureID = [FlutterTexture textureID];
+  FlutterEngineResult success =
+      _flutterEngine.embedderAPI.RegisterExternalTexture(_engine, textureID);
+  if (success == FlutterEngineResult::kSuccess) {
+    _textures[@(textureID)] = FlutterTexture;
+    return textureID;
+  } else {
+    NSLog(@"Unable to register the texture with id: %lld.", textureID);
+    return 0;
+  }
+}
+
+- (void)textureFrameAvailable:(int64_t)textureID {
+  FlutterEngineResult success =
+      _flutterEngine.embedderAPI.MarkExternalTextureFrameAvailable(_engine, textureID);
+  if (success != FlutterEngineResult::kSuccess) {
+    NSLog(@"Unable to mark texture with id %lld as available.", textureID);
+  }
+}
+
+- (void)unregisterTexture:(int64_t)textureID {
+  FlutterEngineResult success =
+      _flutterEngine.embedderAPI.UnregisterExternalTexture(_engine, textureID);
+  if (success == FlutterEngineResult::kSuccess) {
+    [_textures removeObjectForKey:@(textureID)];
+  } else {
+    NSLog(@"Unable to unregister texture with id: %lld.", textureID);
+  }
+}
+
+#pragma mark - Helper methods to create rendering config for embedder.
```

Comment:
This kind of comment belongs on the specific declaration; if anyone adds another private method they are unlikely to notice and update this, at which point it will be actively misleading. You can just use `#pragma mark - Private methods`